models/taskRequests.js

```diff
@@ -0,0 +1,126 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const firestore = require("../utils/firestore");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksModel = require("./tasks");
+const userModel = require("./users");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  const taskRequests = [];
+
+  const visitedUsersId = new Set();
+  const visitedUsers = [];
+
+  const taskRequestsSnapshot = (await taskRequestsCollection.get()).docs;
+
+  for (const taskRequestSnapshot of taskRequestsSnapshot) {
+    const taskRequest = taskRequestSnapshot.data();
+    const requestors = taskRequest.requestors;
+    const users = [];
+
+    const { taskData } = await tasksModel.fetchTask(taskRequest.taskId);
+
+    for (const requestor of requestors) {
+      if (visitedUsersId.has(requestor)) {
+        users.push(visitedUsers.find((visitedUser) => visitedUser.id === requestor));
+        continue;
+      }
+
+      const { user } = await userModel.fetchUser({ userId: requestor });
+      users.push(user);
+      visitedUsersId.add(requestor);
+      visitedUsers.push(user);
+    }
+
+    taskRequests.push({ id: taskRequestSnapshot.id, ...taskRequest, task: taskData, requestors: users });
+  }
+
+  return taskRequests;
+};
+
+/**
+ * Creates a task request
+ *
+ * @param taskId { string }: id of task request
+ * @return {Promise<{taskRequest: Object}>}
+ */
+const addOrUpdate = async (taskId, userId) => {
+  try {
+    const taskRequestsSnapshot = await taskRequestsCollection.where("taskId", "==", taskId).get();
+    const [taskRequestRef] = taskRequestsSnapshot.docs;
```

Comment:
We must do an `isEmpty` check is case there is no document, or a `find()`. Although this code works due `undefined`( thanks to JavaScript (;  )